sda/internal/database/db_functions.go

```diff
@@ -622,3 +622,61 @@ func (dbs *SDAdb) getArchivePath(stableID string) (string, error) {
 
 	return archivePath, nil
 }
+
+// GetUserFiles retrieves all the files a user submitted
+func (dbs *SDAdb) GetUserFiles(userID string) ([]*SubmissionFileInfo, error) {
+	var err error
+
+	files := []*SubmissionFileInfo{}
+
+	// 2, 4, 8, 16, 32 seconds between each retry event.
+	for count := 1; count <= RetryTimes; count++ {
+		files, err = dbs.getUserFiles(userID)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(math.Pow(2, float64(count))) * time.Second)
+	}
+
+	return files, err
+}
```

Comment:
This should have a test case in `db_functions_test.go`.
Preferably testing multiple files with more than 1 status event for each file.

Reply:
see 8caa156